### PR TITLE
Remove dependency on sandboxing host functions

### PR DESCRIPTION
**Pull request by athei**
This removes the dependency on sandboxing host functions by embedding wasmi into the runtime. Previously, it did live inside the client and needed said host functions to communicate with the runtime. This is necessary to use pallet_contracts on a parachain because the sandbox host functions [are not supported there, yet](https://github.com/paritytech/polkadot/issues/99).

With this PR, the sandbox always uses an embedded (compiled into the runtime) wasmi. The only exception is when the experimental `wasmer-sandbox` feature is enabled. In this case the host resident execution engine is used (wasmer in this case). Note that this does not work for the native runtime which will always use an embedded wasmi.

### CODE REVIEW

**Review by cmichi — APPROVED** [review]

File: primitives/sandbox/embedded_executor.rs [hunk 1]
```diff
@@ -15,16 +15,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use sp_std::collections::btree_map::BTreeMap;
-use sp_std::fmt;
-
+use super::{Error, HostError, HostFuncType, ReturnValue, Value, TARGET};
```
Comment (pepyakin) [inline primitives/sandbox/embedded_executor.rs:18]: heh, handy that we haven't killed this.

File: frame/contracts/src/weights.rs [hunk 2]
```diff
@@ -206,11 +206,11 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: Contracts CodeStorage (r:1 w:1)
 	// Storage: Contracts PristineCode (r:0 w:1)
 	fn instantiate_with_code(c: u32, s: u32, ) -> Weight {
-		(182_161_000 as Weight)
-			// Standard Error: 115_000
-			.saturating_add((113_515_000 as Weight).saturating_mul(c as Weight))
-			// Standard Error: 7_000
-			.saturating_add((2_314_000 as Weight).saturating_mul(s as Weight))
+		(479_578_000 as Weight)
```
Comment (pepyakin) [inline frame/contracts/src/weights.rs:209]: 💢

**Review by pepyakin — APPROVED** [review]
Broadly looks fine

File: frame/contracts/src/weights.rs (hunk 2 above)
Comment (athei) [reply frame/contracts/src/weights.rs:209]: Ouch :)

**Comment by athei** [comment]
bot merge

**Pull request merged**